systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java

```diff
@@ -699,6 +708,217 @@ void testKafkaMetricsSettings() {
+        zookeeperLabels.put("replicaId", "$2");
+
+        Map<String, Object> zookeeperRule = new HashMap<>();
+        zookeeperRule.put("labels", zookeeperLabels);
+        zookeeperRule.put("name", "zookeeper_$3");
+        zookeeperRule.put("pattern", "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+)><>(\\w+)");
+
+        Map<String, Object> zookeeperMetrics = new HashMap<>();
+        zookeeperMetrics.put("lowercaseOutputName", true);
+        zookeeperMetrics.put("rules", Collections.singletonList(zookeeperRule));
+
+        String metricsCMNameZk = "zk-metrics-cm";
+        ConfigMap metricsCMZk = new ConfigMapBuilder()
+            .withNewMetadata()
+            .withName(metricsCMNameZk)
+            .withNamespace(testStorage.getNamespaceName())
+            .endMetadata()
+            .withData(singletonMap("metrics-config.yml", mapper.writeValueAsString(zookeeperMetrics)))
+            .build();
+
+        JmxPrometheusExporterMetrics zkMetricsConfig = new JmxPrometheusExporterMetricsBuilder()
+            .withNewValueFrom()
+            .withConfigMapKeyRef(new ConfigMapKeySelectorBuilder()
+                .withName(metricsCMNameZk)
+                .withKey("metrics-config.yml")
+                .withOptional(true)
+                .build())
+            .endValueFrom()
+            .build();
+
+        kubeClient().createConfigMapInNamespace(testStorage.getNamespaceName(), metricsCMK);
+        kubeClient().createConfigMapInNamespace(testStorage.getNamespaceName(), metricsCMZk);
+
+        LOGGER.info("Deploy Kafka Cluster");
+        resourceManager.createResourceWithWait(extensionContext, KafkaTemplates.kafkaEphemeral(testStorage.getClusterName(), 3, 3)
+            .editMetadata()
+            .withNamespace(testStorage.getNamespaceName())
+            .endMetadata()
+            .build());
+
+        // enable metrics and specify kafka exporter
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), kafka -> {
+            kafka.getSpec().getKafka().setMetricsConfig(kafkaMetricsConfig);
+            kafka.getSpec().getZookeeper().setMetricsConfig(zkMetricsConfig);
+            kafka.getSpec().setKafkaExporter(new KafkaExporterSpec());
+
+        }, Environment.TEST_SUITE_NAMESPACE);
+
+        Map<String, String> kafkaPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getKafkaSelector());
+        Map<String, String> zkPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getZookeeperSelector());
+
+        LOGGER.info("Waiting for Kafka and ZooKeeper Pods to roll");
+        zkPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), 3, zkPods);
+        kafkaPods = RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), 3, kafkaPods);
```

Comment:
here again - should we wait for pods ready? Or is it encapsulated somehow? (if so, I guess you should use the `waitTillComponentHasRolledAndPodsReady` for better visibility)